### ISSUE TITLE
Improve UI for validation of self-mapped tasks

### ISSUE THREAD

**Issue by hiddewie** [hotosm/tasking-manager#5088]
From https://github.com/hotosm/tasking-manager/issues/5086#issuecomment-1094172653

**Is your feature request related to a problem? Please describe.**

When a task has been mapped by myself, I cannot validate it. When I select the task and click the *Validate task* button, an error message pops up correctly, preventing me from validating the task I mapped myself. This is good.

**Describe the solution you'd like**

It would be better if the UI showed me that I mapped the task myself, before showing the *Validate task* button, and doesn't let me click the button.

It might show a short explanation below the button why I cannot validate the task.

**Comment by HelNershingThapa** [comment]
Possibly related to https://github.com/hotosm/tasking-manager/issues/3746

**Comment by RAytoun** [comment]
Will changing this also then impact on tasks that a validator has opened for validation and corrected some of the mapping before marking it valid,

**Comment by Ichchhie** [comment]
My personal opinions on this:
Option 1: looks good to me if there is a message on hover like "Mapped by you" as shown below:
![ezgif com-gif-maker (1)](https://user-images.githubusercontent.com/33247422/180741826-6aa7c631-01bc-4eac-b86b-8c83cafdebd9.gif)

Option 2: I feel there is no point giving user to click on the task when the action after click is just showing pop up, also it might confuse if there is multiple tasks selection 

Option 3: There are already 6 colors used in this view, adding new as said can be complex.

**Comment by zlavergne** [comment]
tl;dr: I agree with @Ichchhie's opinions 😃. I think we're already pushing the limits of the color coding system which is why we added the lock icon when we built TM4. I think a tooltip on hover would be the best option of the 3.

One other thought I had was to take the popup one step further and make a general hover popup for any task to show a "condensed" history of the task. I've thought of this in two different ways:
1. same thing as Ichchha's image but we just show whoever did the last action on the task (maybe it was "invalidated by"
2. Show an actual small list of the history. For instance "Mapped by Ichchha", "Invalidated by Petya", "Mapped by Ichchha", "Validated by Aadesh". I would imagine these 3 items would be the longest form we'd need to show. This option is definitely the more utilitarian but could be a bit clunky and goes a bit beyond the original request 😃.

**Comment by hiddewie** [comment]
Huge thanks to all involved, design and implementation, for this feature!

**Comment by Ichchhie** [comment]
> It works good for tasks that I mapped before BUT it behaves the same also if I lock a task for validation and then unlock, I can't click on it anymore. I can still select it from the task list and validate it again but I can't click on the task on the task map.
> 
> ![Image](https://user-images.githubusercontent.com/83399473/192755259-5e0517ce-b504-4ad8-a103-1b0be437d5bd.png)
> 
> ![Image](https://user-images.githubusercontent.com/83399473/192755336-d625fbad-53b1-41f2-a064-92ba0b38fd0c.png)

Thank you @Patrik-Br for testing so well. We replicated the issue you mentioned. Will need to work on it and let you know once the issue is fixed.
